sklearn/utils/__init__.py

```diff
@@ -983,6 +984,27 @@ def get_chunk_n_rows(row_bytes, *, max_n_rows=None, working_memory=None):
     return chunk_n_rows
 
 
+def _is_pandas_na(x):
+    """Test if x is pandas.NA.
+
+    Parameters
+    ----------
+    x : any type
+
+    Returns
+    -------
+    boolean
+    """
+    with suppress(ImportError, AttributeError):
+        # We also suppress `AttributeError` because older versions of pandas do
+        # not have `NA`.
+        import pandas
+
+        return x is pandas.NA
+
+    return False
+
+
 def is_scalar_nan(x):
```

Comment:
Is the fact that is not used internally by `is_scalar_nan` intentional because estimators that support `pd.NA` are the exception rather than the rule for now and that each time an estimator supports `pd.NA` specific code has to be written?

Reply:
Right, it is non-trivial to support `pd.NA` everywhere, for example:

``` python
import numpy as np
import pandas
from sklearn.utils.validation import check_array
check_array([0.1, pandas.NA], dtype=np.float64)

----------------
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/yxiong/src/scikit-learn/sklearn/utils/validation.py", line 739, in check_array
    array = np.asarray(array, order=order, dtype=dtype)
TypeError: float() argument must be a string or a number, not 'NAType'
```

This PR just gets the `SimpleImputer` right, as a starting point.

Reply:
Maybe you could add an explicit comment to explain that we intentionally not check for `is_pandas_na` in `is_scalar_nan` because in some cases, supporting `pandas.NA` requires specific adaptation in the estimator itself that needs to be done on a case by case basis.

Reply:
Done.